src/torrent.cpp

```diff
@@ -6797,10 +6806,13 @@ namespace libtorrent
 				}
 #endif
 				p->disconnect(errors::too_many_connections, op_bittorrent);
-				// we have to do this here because from the peer's point of
+				// we have to do this here because from the peer's point of view
 				// it wasn't really attached to the torrent, but we do need
 				// to let peer_list know we're removing it
 				remove_peer(p);
+				// the peer is already inserted and needs to be removed
+				TORRENT_ASSERT(sorted_find(m_connections, p) != m_connections.end());
+				m_connections.erase(sorted_find(m_connections, p));
```

Comment:
yes, I misread the context here. I think it would be nicer if remove_peer() would do this also